core/src/main/java/org/radargun/Slave.java

```diff
@@ -46,6 +49,8 @@ private void run(int slaveIndex) throws Exception {
          if (object == null) {
             log.info("Master shutdown!");
             break;
+         } else if (object instanceof ArrayList) {
```

Comment:
Eh, this is rather generic object... what about adding the addresses into `Cluster` instead? Or at least add a new class that distinguishes the data.